stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+                // Leap second happens at ls._Hour:59:59. We store the next second after, so add a whole hour.
+                const auto& _Ls = _Reg_ls_data[_Idx];
+                const auto _Date =
+                    static_cast<sys_days>(year_month_day{year{_Ls._Year}, month{_Ls._Month}, day{_Ls._Day}})
+                    + hours{_Ls._Hour + 1};
+                _Leap_sec_info.emplace_back(_Date, !_Ls._Negative);
+                _All_ls_positive = _All_ls_positive && !_Ls._Negative;
+            }
+        }
+
+        return {_Leap_sec_info, _All_ls_positive};
+    }
+
+    // [time.zone.db]
+
+    // TRANSITION: work in progress
+    struct tzdb {
+        vector<leap_second> leap_seconds;
+        bool _All_ls_positive;
+    };
+
+    // TRANSITION: work in progress
+    class tzdb_list {
+    private:
+        using _ListType = forward_list<tzdb>;
+
+    public:
+        using const_iterator = _ListType::const_iterator;
+
+        tzdb_list(const tzdb_list&) = delete;
+        tzdb_list& operator=(const tzdb_list&) = delete;
+        tzdb_list()                            = default;
+
+        const tzdb& front() const noexcept {
+            _Shared_lock _Lk(_Tzdb_mutex);
+            return _Tzdb_list.front();
+        }
+
+        template <class... _ArgsTy>
+        void _Emplace_front(_ArgsTy&&... _Args) {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            _Tzdb_list.emplace_front(_STD forward<_ArgsTy>(_Args)...);
+        }
+
+        const tzdb& _Reload() {
+            _Unique_lock _Lk(_Tzdb_mutex);
+            auto [_Leap_sec, _All_ls_positive] = _Xtzdb_generate_leap_seconds(_Tzdb_list.front().leap_seconds.size());
+            if (!_Leap_sec.empty()) {
+                _Emplace_front(tzdb{_STD move(_Leap_sec), _All_ls_positive});
+            }
+            return front();
+        }
+
+        void _Initialize() {
+            _Shared_lock _Shr_lk(_Tzdb_mutex);
+            if (_Tzdb_list.empty()) {
+                _Shr_lk._Unlock();
+                _Unique_lock _Lk(_Tzdb_mutex);
+                if (_Tzdb_list.empty()) {
+                    auto _Leap_sec = _Xtzdb_generate_leap_seconds(0);
+                    _Tzdb_list.emplace_front(tzdb{_STD move(_Leap_sec.first), _Leap_sec.second});
+                }
+            }
+        }
+
+    private:
+        _ListType _Tzdb_list;
+        mutable _Smtx_t _Tzdb_mutex = {};
+
+        struct _NODISCARD _Shared_lock {
+            explicit _Shared_lock(_Smtx_t& _Mtx_) : _Mtx{&_Mtx_} {
+                _Smtx_lock_shared(_Mtx);
+            }
+
+            void _Unlock() {
+                if (owns) {
+                    _Smtx_unlock_shared(_Mtx);
+                }
+                owns = false;
+            }
+
+            ~_Shared_lock() {
+                if (owns) {
+                    _Smtx_unlock_shared(_Mtx);
+                }
+            }
+
+            _Smtx_t* _Mtx;
+            bool owns = true;

```

Comment:
I believe that `owns` should be `_Ugly` as this is an internal type. (There are only a few identifiers that we feel "safe" using as non-`_Ugly`, like `type`, `value`, and `pointer`.)